amalgamation.sh

```diff
@@ -100,80 +105,29 @@ echo "/* auto-generated on ${timestamp}. Do not edit! */" > "${AMAL_H}"
 echo "Creating ${AMAL_C}..."
 echo "/* auto-generated on ${timestamp}. Do not edit! */" > "${AMAL_C}"
 {
-    echo "#include \"${AMAL_H}\""
-
-    echo ""
-    echo "/* used for http://dmalloc.com/ Dmalloc - Debug Malloc Library */"
-    echo "#ifdef DMALLOC"
-    echo "#include \"dmalloc.h\""
-    echo "#endif"
-    echo ""
-
     for file in ${ALLCFILES}; do
         dofile "$SRCPATH/$file"
     done
 } >> "${AMAL_C}"
 
-
-DEMOCPP="amalgamation_demo.cpp"
-echo "Creating ${DEMOCPP}..."
-echo "/* auto-generated on ${timestamp}. Do not edit! */" > "${DEMOCPP}"
-cat <<< '
-#include <iostream>
-#include "simdjson.h"
-#include "simdjson.cpp"
-int main(int argc, char *argv[]) {
-  if(argc < 2) {
-    std::cerr << "Please specify at least one file name. " << std::endl;
-  }
-  const char * filename = argv[1];
-  simdjson::dom::parser parser;
-  auto [doc, error] = parser.load(filename); // do the parsing
-  if (error) {
-    std::cout << "parse failed" << std::endl;
-    std::cout << "error code: " << error << std::endl;
-    std::cout << error << std::endl;
-  } else {
-    std::cout << "parse valid" << std::endl;
-  }
-  if(argc == 2) {
-    return EXIT_SUCCESS;
-  }
-
-  // parse_many
-  const char * filename2 = argv[2];
-  for (auto result : parser.load_many(filename2)) {
-    error = result.error();
-  }
-  if (error) {
-    std::cout << "parse_many failed" << std::endl;
-    std::cout << "error code: " << error << std::endl;
-    std::cout << error << std::endl;
-  } else {
-    std::cout << "parse_many valid" << std::endl;
-  }
-  return EXIT_SUCCESS;
-}
-' >>  "${DEMOCPP}"
-
 echo "Done with all files generation. "
-
 echo "Files have been written to  directory: $PWD "
-ls -la ${AMAL_C} ${AMAL_H}  ${DEMOCPP}
+ls -la ${AMAL_C} ${AMAL_H}
 
 echo "Giving final instructions:"
 
 
+DEMOCPP="amalgamation_demo.cpp"
 CPPBIN=${DEMOCPP%%.*}
+SINGLEHDR=$SCRIPTPATH/singleheader
 
 echo "Try :"
 echo "c++ -O3 -std=c++17 -pthread -o ${CPPBIN} ${DEMOCPP}  && ./${CPPBIN} ../jsonexamples/twitter.json ../jsonexamples/amazon_cellphones.ndjson"
 
-SINGLEHDR=$SCRIPTPATH/singleheader
-echo "Copying files to $SCRIPTPATH/singleheader "
+echo "Copying files to $SINGLEHDR"
 mkdir -p $SINGLEHDR
 echo "c++ -O3 -std=c++17 -pthread -o ${CPPBIN} ${DEMOCPP}  && ./${CPPBIN} ../jsonexamples/twitter.json ../jsonexamples/amazon_cellphones.ndjson" > $SINGLEHDR/README.md
-cp ${AMAL_C} ${AMAL_H}  ${DEMOCPP} $SINGLEHDR
+cp ${AMAL_C} ${AMAL_H} $SINGLEHDR
```

Comment:
I'm actually super OK with this, I just blithely kept the copy out of momentum. @lemire are we OK with not putting amalgamation_demo into the produced library anymore?

We should remove the amalgamation_demo.cpp from singleheader as well, if so.

Reply:
> We should remove the amalgamation_demo.cpp from singleheader as well, if so.


I don't understand. This makes it so that the script does not create the demo program. If we do not include it, how are we going to run checks on the single-header code? Of course, we could compile some other piece of code.

Reply:
The demo program is always constant. script shouldn't generate it from the start. In this PR there are two commits, one removes that generation of amalgamation_demo.cpp, perhaps, it should later be moved to examples or some other folder out of singleheader dir